routes/api/v1/playlists.js

```diff
@@ -57,95 +63,78 @@ router.put('/:id', async function (request, response) {
     return response.status(500).json({"error": "Request could not be handled"});
   }
 });
-//
-router.delete('/:id', async function (request, response) {
-  try {
-    let playlistId = await request.params.id
-    let data = await playlists.findPlaylist(playlistId)
 
-    if (data.length != 0){
-      await playlists.deletePlaylist(data[0].id)
-      return response.status(204).json(data);
-    } else {
-      return response.status(404).json({"error": "Record not found"});
-    }
-  }
-  catch(error) {
-    return response.status(500).json({"error": "Request could not be handled"});
-  }
-});
-//
 
-router.delete('/:playlist_id/favorites/:favorite_id', async function (request, response) {
+
+router.post('/:playlist_id/favorites/:favorite_id', async function (request, response) {
   try {
-    let playlistId = await request.params.playlist_id
     let favoriteId = await request.params.favorite_id
-    let data = await playlists.removeFavoriteFromPlaylist(playlistId, favoriteId);
-    if (data){
-      return response.status(204).json(data);
-    } else{
-      return response.status(404).json({"error": "Record not found"})
+    let playlistId = await request.params.playlist_id
+    let previouslyfavorited = await playlistsFavorites.allPlaylistsFavorites(favoriteId, playlistId)
+    if(previouslyfavorited.length != []) {
+      return response.status(400).json({"error": "The song has already been added to the playlist"});
+    }
+    let newFavorited = await playlistsPresenter.getPlaylistFavoriteTitle(favoriteId, playlistId)
+    if (newFavorited){
+      await playlists.addFavoriteToPlaylist(favoriteId, playlistId)
+      return response.status(201).json({"Success": `${newFavorited.favorite} has been added to ${newFavorited.playlist}!`})
+    } else {
+      return response.status(400).json({"error": "Please enter a valid playlist and/or favorite id"});
     }
   }
   catch(error) {
     return response.status(500).json({"error": "Request could not be handled"});
   }
 });
 
-router.post('/', async function (request, response) {
+
+router.put('/:id', async function (request, response) {
+  let playlistId = await request.params.id
+  let title = await request.body.title
+  let playlist = await playlists.findPlaylist(playlistId)
   try {
+    if (playlist.length == 0) {
+      return response.status(400).json({"error": "Please enter a valid id"});
+    }
     if (!('title' in request.body)) {
       return response.status(400).json({"error": "You must include a title parameter in the request"});
     }
     if (request.body.title === '') {
       return response.status(400).json({"error": "Title cannot be blank"});
     }
-    try {
```

Comment:
Good idea here with taking out the nested try/catches. This will allow users to now see a server error in those instances